model/testing/src/icon4py/model/testing/fixtures/datatest.py

```diff
@@ -108,12 +115,56 @@ def processor_props(request):
 
 
 @pytest.fixture(scope="session")
-def ranked_data_path(processor_props):
-    return dt_utils.get_ranked_data_path(dt_utils.SERIALIZED_DATA_PATH, processor_props)
+def ranked_data_path(processor_props: decomposition.ProcessProperties) -> pathlib.Path:
+    return dt_utils.get_ranked_data_path(
+        definitions.serialized_data_path(), processor_props.comm_size
+    )
+
+
+def _download_ser_data(
+    comm_size: int,
+    _ranked_data_path: pathlib.Path,
+    _experiment: str,
+):
+    # not a fixture to be able to use this function outside of pytest
+    try:
+        destination_path = dt_utils.get_datapath_for_experiment(_ranked_data_path, _experiment)
+        if _experiment == dt_utils.GLOBAL_EXPERIMENT:
+            uri = dt_utils.DATA_URIS_APE[comm_size]
+        elif _experiment == dt_utils.JABW_EXPERIMENT:
+            uri = dt_utils.DATA_URIS_JABW[comm_size]
+        elif _experiment == dt_utils.GAUSS3D_EXPERIMENT:
+            uri = dt_utils.DATA_URIS_GAUSS3D[comm_size]
+        elif _experiment == dt_utils.WEISMAN_KLEMP_EXPERIMENT:
+            uri = dt_utils.DATA_URIS_WK[comm_size]
+        else:
+            uri = dt_utils.DATA_URIS[comm_size]
+
+        data_file = _ranked_data_path.joinpath(f"{_experiment}_mpitask{comm_size}.tar.gz").name
+        _ranked_data_path.mkdir(parents=True, exist_ok=True)
+        with locking.lock(_ranked_data_path):
```

Comment:
Now, the lock should take care of only one of the mpi ranks downloading the data.